File: Task_1/task_1.kt
fun isCyclicShift(s: String, t: String): Boolean {
    if (s.length != t.length) { // Проверяем, равны ли длины строк
        return false
    }

    val doubledS = s + s // Создаем строку S+S

    return doubledS.contains(t) // Проверяем, содержится ли T в S+S
}

fun main() {
    // Считываем строки S и T с клавиатуры
    println("Enter string S:")
    val s = readLine() ?: return println("Error reading string S.")

    println("Введите строку T:")
    val t = readLine() ?: return println("Error reading string T.")

    if (isCyclicShift(s, t)) { // Проверяем, является ли T циклическим сдвигом S
        println("Yes")
    } else {
        println("No")
    }
}

File: Task_1/task_1.ts
function isCyclicShift(s: string, t: string): boolean {
    // Проверяем, равны ли длины строк
    if (s.length !== t.length) {
        return false;
    }

    // Создаем строку S+S
    const doubledS = s + s;

    // Проверяем, содержится ли T в S+S
    return doubledS.includes(t);
}

async function main() {
    // Считываем строки S и T с помощью prompt
    const s = prompt("Введите строку S:");
    if (!s) {
        alert("Ошибка при вводе строки S.");
        return;
    }

    const t = prompt("Введите строку T:");
    if (!t) {
        alert("Ошибка при вводе строки T.");
        return;
    }

    // Проверяем, является ли T циклическим сдвигом S
    if (isCyclicShift(s, t)) {
        alert("Yes");
    } else {
        alert("No");
    }
}

// Запускаем программу
main();